CONTRIBUTING.md

```diff
@@ -0,0 +1,73 @@
+# Contributing to the Lua Users Foundation
+
+This document serves as contribution guidelines and the points listed should be
+followed for contributions to the foundation.
+
+## Commit Guidelines
+
+As is somewhat standard practice, the following should be observed:
+
+- The usage of Markdown-formatted lists is allowed
+- Paragraphs should be broken with an empty line between
+- Further lines after the header should be no more than 72 characters
+- Further lines, if required, should contain detailed information on:
+ 1. Implications of the changes in the commit
+ 2. Reasons the changes are required
+
+Commits relating to one file should follow these guidelines:
+
+- The header line should be no more than 60 characters
+- The header line should contain the filename and a short description of what
+applying the patch will do, in the format of "If applied, this commit will..."
+- The second line should be blank
+
+Example commit message:
+
+```markdown
+CONTRIBUTING.md: Establish guidelines for new commits
+
+Implications of Changes:
+
+- Could possibly make it harder for outsiders to contribute
+- Could create bloatware in the repository data
+
+Reasons for Changes:
+
+- Creates a structure which can be used to avoid excessive comments
+  during the review process
+```
+
+Commits relating to multiple files should follow these guidelines:
+
+- The header line should be no more than 50 characters
+- The header line should contain a short description of what applying the patch
+will do, in the format of "If applied, this commit will..."
+- The second line should be blank
+- The third line should begin a list of changed files, with a header along the
+lines of "Changed Files:"
+
+Example commit message:
+
+```markdown
+Apply changes to files as per guidelines
+
+Changed Files:
```

Comment:
Why is this needed? Doesn't git itself provide this outside of the commit message? Secondly, most this repo will consist of documents, not code, I don't think we need to be very strict with the body of the message as long as the header is clear. Or that some description is provided somewhere, such as in the PR description. To be frank, I would rather have a longer description at the PR than at each commit.

Reply:
Doesn't GitHub auto-populate the PR with the contents of the commit?  My goal here is to avoid someone finding a segment of documentation that was added however long ago, and having to look back for a closed PR instead of being able to look directly at the commit.

Reply:
I'm not sure if github does that? This PR has empty description

Reply:
As discussed on Slack, it puts the first line in as the title, and the rest of the lines - which in this case didn't exist - as the description.